### ISSUE TITLE
[Bug]: Matplotlib crashes Python

### ISSUE THREAD

**Issue by juhaszp95** [matplotlib/matplotlib#21511]
### Bug summary

When you invoke pyplot.show() to show your plot, the plot is not shown but instead crashes Python. This came about when I updated the package 'freetype' in a conda installation, from version 2.10.4 to 2.11.0.

### Code for reproduction

```python
import matplotlib.pyplot as plt
plt.plot(1,1)
plt.show()
```

### Actual outcome

Python simply crashes with no error messages.

### Expected outcome

It should plot a single point at (1,1) and show that plot.

### Operating system

Windows 10

### Matplotlib Version

3.4.3

### Matplotlib Backend

Qt5Agg

### Python version

3.8.12

### Jupyter version

N/A

### Other libraries

The bug came about when I updated package freetype, from version 2.10.4 to 2.11.0. The output of conda list (before the update):

packages in environment at C:\juhasz-local\Anaconda3\envs\plttest:

Name                    Version                   Build  Channel
blas                      1.0                         mkl
blosc                     1.21.0               h19a0ad4_0
brotli                    1.0.9                ha925a31_2
bzip2                     1.0.8                he774522_0
ca-certificates           2021.10.26           haa95532_2
certifi                   2021.10.8        py38haa95532_0
cfitsio                   3.470                he774522_6
charls                    2.2.0                h6c2663c_0
cycler                    0.10.0                   py38_0
fonttools                 4.25.0             pyhd3eb1b0_0
freetype                  2.10.4               hd328e21_0
giflib                    5.2.1                h62dcd97_0
icu                       58.2                 ha925a31_3
imagecodecs               2021.8.26        py38ha1f97ea_0
intel-openmp              2021.3.0          haa95532_3372
jpeg                      9d                   h2bbff1b_0
kiwisolver                1.3.1            py38hd77b12b_0
lcms2                     2.12                 h83e58a3_0
lerc                      3.0                  hd77b12b_0
libaec                    1.0.4                h33f27b4_1
libdeflate                1.8                  h2bbff1b_5
libpng                    1.6.37               h2a8f88b_0
libtiff                   4.2.0                hd0e1b90_0
libwebp                   1.2.0                h2bbff1b_0
libzopfli                 1.0.3                ha925a31_0
lz4-c                     1.9.3                h2bbff1b_1
matplotlib                3.4.3            py38haa95532_0
matplotlib-base           3.4.3            py38h49ac443_0
mkl                       2021.3.0           haa95532_524
mkl-service               2.4.0            py38h2bbff1b_0
mkl_fft                   1.3.1            py38h277e83a_0
mkl_random                1.2.2            py38hf11a4ad_0
munkres                   1.1.4                      py_0
numpy                     1.21.2           py38hfca59bb_0
numpy-base                1.21.2           py38h0829f74_0
olefile                   0.46               pyhd3eb1b0_0
openjpeg                  2.4.0                h4fc8c34_0
openssl                   1.1.1l               h2bbff1b_0
pillow                    8.4.0            py38hd45dc43_0
pip                       21.2.2           py38haa95532_0
pyparsing                 2.4.7              pyhd3eb1b0_0
pyqt                      5.9.2            py38ha925a31_4
python                    3.8.12               h6244533_0
python-dateutil           2.8.2              pyhd3eb1b0_0
qt                        5.9.7            vc14h73c81de_0
setuptools                58.0.4           py38haa95532_0
sip                       4.19.13          py38ha925a31_0
six                       1.16.0             pyhd3eb1b0_0
snappy                    1.1.8                h33f27b4_0
sqlite                    3.36.0               h2bbff1b_0
tk                        8.6.11               h2bbff1b_0
tornado                   6.1              py38h2bbff1b_0
vc                        14.2                 h21ff451_1
vs2015_runtime            14.27.29016          h5e58377_2
wheel                     0.37.0             pyhd3eb1b0_1
wincertstore              0.2              py38haa95532_2
xz                        5.2.5                h62dcd97_0
zfp                       0.5.5                hd77b12b_6
zlib                      1.2.11               h62dcd97_4
zstd                      1.4.9                h19a0ad4_0

### Installation

conda

### Conda channel

default

**Comment by Mental59** [comment]
I had the same problem, i just removed matplotlib with conda uninstall matplotlib and then install it with command pip install matplotlib.

**Comment by trenton3983** [comment]
Please see [SO: python kernel crashes when plotting with matplotlib](https://stackoverflow.com/q/69786885/7758804)

**Comment by trenton3983** [comment]
Issue confirmed independently, prior to finding this bug report. [python kernel crashes when plotting with matplotlib after conda update](https://stackoverflow.com/q/69786885/7758804)

**Comment by QuLogic** [comment]
If downgrading FreeType from conda-forge fixes it, then you should report a bug to the conda-forge feedstock: https://github.com/conda-forge/freetype-feedstock

If it's FreeType from anaconda, then you should report it to them, but I don't know where that is.

**Comment by juhaszp95** [comment]
I see it's just been reported for conda-forge. Actually, it doesn't matter which channel freetype (or matolotlib) is installed from, it crashes regardless whether installed from default or conda-forge.

**Comment by ericpre** [comment]
To clarify, freetype 2.11.0 is not on conda-forge yet: https://anaconda.org/conda-forge/freetype. The PR to is add 2.11.0 on conda-forge is pending: https://github.com/conda-forge/freetype-feedstock/pull/41.

It should be coming from anaconda main/defaults channels: https://anaconda.org/anaconda/freetype

**Comment by ericpre** [comment]
I was able to reproduce on windows but not linux with the minimum reproducible example given in https://github.com/AnacondaRecipes/freetype-feedstock/issues/3.

**Comment by Dragon1573** [comment]
## ⚠️ Bug summary

When I invoke `plt.show()`, it crashes the whole Python Intepreter instead of showing up the figure correctly.

### Code to reproduce

```python
from matplotlib import pyplot as plt
plt.figure(fig_size=(4, 3), dpi=600)
plt.plot([1, 2, 3], [1, 4, 9])
plt.show()
```

### Actual outcome

Python crashed without showing any errors or stacktraces. 

### Expected outcome

`matplotlib.pyplot` should plot a figure and display it for me.

## Environments

<details>
<summary>Operating System</summary>
<p>Windows 10 Pro 21H1 (19043.1320)</p>
</details>


<details>
  <summary>Matplotlib Version</summary>
  <code>
<pre>(test) ~ λ conda list matplotlib
# packages in environment at E:\Applications\Miniconda3\envs\test:
#
# Name                    Version                   Build  Channel
matplotlib                3.4.3            py37haa95532_0    main
matplotlib-base           3.4.3            py37h49ac443_0    main</pre>
  </code>
</details>


<details>
  <summary>Matplotlib Backend</summary>
  <code>
<pre>(test) ~ λ python -c '
>> from matplotlib import pyplot as plt;
>> print(plt.get_backend());
>> '
Qt5Agg</pre>
  </code>
</details>

<details>
  <summary>Python Version</summary>
  <code>
<pre>(test) ~ λ conda list python
# packages in environment at E:\Applications\Miniconda3\envs\test:
#
# Name                    Version                   Build  Channel
python                    3.7.11               h6244533_0    main
python-dateutil           2.8.2              pyhd3eb1b0_0    main
(test) ~ λ python --version
Python 3.7.11</pre>
  </code>
</details>

<details>
  <summary>Jupyter Version</summary>
  <p>
    Not installed ...
  </p>
</details>

<details>
  <summary>Other libraries in Conda</summary>
  <code>
<pre>(test) ~ λ conda list -n test
# packages in environment at E:\Applications\Miniconda3\envs\test:
#
# Name                    Version                   Build  Channel
blas                      1.0                         mkl    main
brotli                    1.0.9                ha925a31_2    main
ca-certificates           2021.10.26           haa95532_2    main
certifi                   2021.10.8        py37haa95532_0    main
cycler                    0.10.0                   py37_0    main
fonttools                 4.25.0             pyhd3eb1b0_0    main
freetype                  2.11.0               ha860e81_0    main
icu                       58.2                 ha925a31_3    main
intel-openmp              2021.4.0          haa95532_3556    main
jpeg                      9d                   h2bbff1b_0    main
kiwisolver                1.3.1            py37hd77b12b_0    main
libpng                    1.6.37               h2a8f88b_0    main
libtiff                   4.2.0                hd0e1b90_0    main
libwebp                   1.2.0                h2bbff1b_0    main
lz4-c                     1.9.3                h2bbff1b_1    main
matplotlib                3.4.3            py37haa95532_0    main
matplotlib-base           3.4.3            py37h49ac443_0    main
mkl                       2021.4.0           haa95532_640    main
mkl-service               2.4.0            py37h2bbff1b_0    main
mkl_fft                   1.3.1            py37h277e83a_0    main
mkl_random                1.2.2            py37hf11a4ad_0    main
munkres                   1.1.4                      py_0    main
numpy                     1.21.2           py37hfca59bb_0    main
numpy-base                1.21.2           py37h0829f74_0    main
olefile                   0.46                     py37_0    main
openssl                   1.1.1l               h2bbff1b_0    main
pillow                    8.4.0            py37hd45dc43_0    main
pip                       21.2.4           py37haa95532_0    main
pyparsing                 3.0.4              pyhd3eb1b0_0    main
pyqt                      5.9.2            py37h6538335_2    main
python                    3.7.11               h6244533_0    main
python-dateutil           2.8.2              pyhd3eb1b0_0    main
qt                        5.9.7            vc14h73c81de_0    main
setuptools                58.0.4           py37haa95532_0    main
sip                       4.19.8           py37h6538335_0    main
six                       1.16.0             pyhd3eb1b0_0    main
sqlite                    3.36.0               h2bbff1b_0    main
tk                        8.6.11               h2bbff1b_0    main
tornado                   6.1              py37h2bbff1b_0    main
vc                        14.2                 h21ff451_1    main
vs2015_runtime            14.27.29016          h5e58377_2    main
wheel                     0.37.0             pyhd3eb1b0_1    main
wincertstore              0.2              py37haa95532_2    main
xz                        5.2.5                h62dcd97_0    main
zlib                      1.2.11               h62dcd97_4    main
zstd                      1.4.9                h19a0ad4_0    main</pre>
  </code>
</details>

<details>
  <summary>Installation</summary>
  <p>
    Using conda version 4.10.3
  </p>
</details>

<details>
  <summary>Conda Channels</summary>
  <code>
<pre>(test) ~ λ conda config --show-sources
==> C:\Users\Dragon1573\.condarc <==
auto_activate_base: True
changeps1: True
channels:
  - main
  - r
  - pkgs/msys2
  - conda-forge
  - cloud/msys2
  - bioconda
  - menpo
  - pytorch
  - simpleitk
  - plotly
custom_channels:
  main: https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs
  r: https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs
  pkgs/msys2: https://mirrors.tuna.tsinghua.edu.cn/anaconda
  conda-forge: https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud
  cloud/msys2: https://mirrors.tuna.tsinghua.edu.cn/anaconda
  bioconda: https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud
  menpo: https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud
  pytorch: https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud
  simpleitk: https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud
  plotly: https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud
show_channel_urls: True</pre>
  </code>
</details>

**Comment by Dragon1573** [comment]
@trenton3983, 

I really appreciate your help! Downgrade the version of `freetype` from `main/freetype::2.11.0` to `main/freetype::2.10.4` does solve the problem 🎉

Now I create a file in `${CONDA_ENV_HOME}/conda-meta/pinned` and configure `freetype=2.10.*` to prevent it from upgrading.

**Comment by dopplershift** [comment]
Another option if you have `conda-forge` as your highest priority channel (top of the list) in your `.condarc` file is to do:
`conda config --set channel_priority strict`. This will prioritize conda-forge's version of packages, regardless of a newer version being available from defaults. It also has the benefit of making package resolution for install/env creation somewhat faster.

**Comment by dopplershift** [comment]
Installing Anaconda's freetype 2.11.0 into my otherwise conda-forge environment on macOS was not enough to reproduce the crash.

**Comment by bluehobo** [comment]
This problem is the sort of thing that will make educators sceptical to use Python as a platform to teach code in schools. Fix it now.

**Comment by tacaswell** [comment]
@bluehobo A vast majority the the contributors to Matplotlib and the scipy eco-system at large are volunteers doing this on their own time.  Please keep your comments professional and constructive.

https://github.com/matplotlib/matplotlib/blob/main/CODE_OF_CONDUCT.md

**Comment by bluehobo** [comment]
> @bluehobo A vast majority the the contributors to Matplotlib and the scipy eco-system at large are volunteers doing this on their own time. Please keep your comments professional and constructive.
> 
> https://github.com/matplotlib/matplotlib/blob/main/CODE_OF_CONDUCT.md

I'm sorry, and didnt mean to step on any toes. However, for most users of Anaconda now, Matplotlib hasnt't been working for days. And most of the users in the education sector do not possess the skill to search the internet for a manual downgrade fix, as a coder would. Downgrading the freetype library worked for me if that's any help.

**Comment by chenghlee** [comment]
FYI: Anaconda is [soft yanking](https://github.com/AnacondaRecipes/repodata-hotfixes/pull/150) the Windows freetype 2.11.0 packages on defaults until we've figured out exactly what's going on. We've confirmed the matplotlib compatibility issue is Windows-specific; we decided to yank the package because we don't know if the root cause also impacts other freetype-dependents on Windows.

**Comment by tacaswell** [comment]
Thanks @chenghlee !  I'm going to close this as there is nothing to be done on the Matplotlib side, hopefully the issues with freetype gets sorted out quickly!  Please let us know if you need any help testing.

**Comment by dopplershift** [comment]
Thanks @chenghlee !

**Comment by juhaszp95** [comment]
I'm sorry but how can we be sure the issue is within the freetype package, and not matolotlib? Maybe matolotlib is not handling some new feature of freetype well? Or at least, maybe the dependencies of matolotlib need to be altered to say freetype 2.11.0 is not compatible?

**Comment by dopplershift** [comment]
@juhaszp95 We don't know that there's not something wrong. All we know is that on Windows, 2.10.4 doesn't result in crashes but 2.11.0. As noted by @chenghlee
> we decided to yank the package because we don't know if the root cause also impacts other freetype-dependents on Windows.

Yanking the 2.11.0 release is the expedient way to make sure lots of users don't encounter the crash while people have time to investigate and determine the root cause of the problem.

**Comment by tacaswell** [comment]
The issue is only on Windows and Fedora has been building with FT2.11 for a while.  That it works on other platfroms gives me confidence that the API has not changed under us.  Based on the reports, my suspicions are first a build issue with FT2.11 in the Anaconda main channel on windows where some c-library has gotten crossed and second a regression in Freetype on windows.  

I think conda-forge is making progress on building FT2.11 in their system, assuming someone with a Windows machine can get a hold of those built packages (with out publishing them to the main CF channel!) and they work then we can narrow this down to a build issue with anaconda's package (and if they also are broken could still be a build issue or a freetype on windows regression). 

Given that Anaconda has yanked the bad binaries and are looking into it, we should wait for their findings before doing anything drastic.

**Comment by tacaswell** [comment]
well, @dopplershift and I have slightly different debugging decision trees :-p

**Comment by dopplershift** [comment]
@tacaswell Well I agree with your debugging decision tree, just more arguing that yanking the problematic release is the expedient and kindest solution for users while some kind of debugging process is engaged--one that doesn't negatively impact users.

**Comment by juhaszp95** [comment]
I see. But then the bug still exists, so maybe it's worth keeping this ticket open? (Maybe that's not standard practise when the problem has been 'solved' by yanking freetype.)

**Comment by jklymak** [comment]
We are 98% sure this is not a matplotlib issue as outlined by @tacaswell. If further investigation turns up a problem with matplotlib we can reopen

You can check this yourself by trying to use freetype without matplotlib and seeing if it works.  If freetype seems fully functional please do report back to us!

**Comment by tarekzoabi19** [comment]
I have downgrades to  freetype 2.10.4 and I still have the issue, is there anything else I should do?

**Comment by trenton3983** [comment]
> I downgraded to freetype 2.10.4 and I still have the issue, is there anything else I should do?

Make sure you have downgraded `freetype` in the correct environment.

**Comment by MatthijsdeJ** [comment]
Downgrading `freetype` did not solve the problem for me. What did was:

`conda update intel-openmp`

Found [here](https://stackoverflow.com/questions/59764844/kernel-keeps-dying-and-restarting-when-running-some-lines-in-python-notebook).

**Comment by specter119** [comment]
> Downgrading `freetype` did not solve the problem for me. What did was:
> 
> `conda update intel-openmp`
> 
> Found [here](https://stackoverflow.com/questions/59764844/kernel-keeps-dying-and-restarting-when-running-some-lines-in-python-notebook).

I think is another issue (far from this), some of my conda envs even don't have the `intel-openmp` packages.

**Comment by tacaswell** [comment]
@specter119 Can you try with https://docs.python.org/3/library/faulthandler.html turned on?

**Comment by daigz1224** [comment]
> I downgraded to freetype 2.10.4 and I still have the issue, is there anything else I should do?

Same. I solved it by reinstall miniconda.

**Comment by Dzungarian** [comment]
By the way, there is no issue with Jupyter Notebook in VS Code but browser based Jupyter Notebook is not functioning.

**Comment by tacaswell** [comment]
> By the way, there is no issue with Jupyter Notebook in VS Code but browser based Jupyter Notebook is not functioning.

This is almost certainly because they are looking at different (conda) environments and interpreters.

**Comment by Dzungarian** [comment]
> > By the way, there is no issue with Jupyter Notebook in VS Code but browser based Jupyter Notebook is not functioning.
> 
> This is almost certainly because they are looking at different (conda) environments and interpreters.

The use the same interpreter.

**Comment by fontanads** [comment]
Maybe it is outdated, but I've tried some suggestions without success.
Finally, updating matplotlib as `conda install matplotlib=3.6.1 -c conda-forge` worked and solved the problem (it was 3.5.2 in my env with the issue).

**Comment by SajeewaGarusinghe** [comment]
> @trenton3983,
> 
> I really appreciate your help! Downgrade the version of `freetype` from `main/freetype::2.11.0` to `main/freetype::2.10.4` does solve the problem 🎉
> 
> Now I create a file in `${CONDA_ENV_HOME}/conda-meta/pinned` and configure `freetype=2.10.*` to prevent it from upgrading.

this doesnt work for me ,
btw following worked.

#this code block is for solving matplotlib kernel dieing issue
'''
import os
os.environ["KMP_DUPLICATE_LIB_OK"]="TRUE"
'''
